flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/SchedulerBase.java

```diff
@@ -845,4 +854,58 @@ private String retrieveTaskManagerLocation(ExecutionAttemptID executionAttemptID
 			.orElse("Unknown location");
 	}
 
+	@Override
+	public void deliverOperatorEventToCoordinator(
+			final ExecutionAttemptID taskExecutionId,
+			final OperatorID operatorId,
+			final OperatorEvent evt) throws FlinkException {
+
+		// Failure semantics (as per the javadocs of the method):
+		// If the task manager sends an event for a non-running task or an non-existing operator
+		// coordinator, then respond with an exception to the call. If task and coordinator exist,
+		// then we assume that the call from the TaskManager was valid, and any bubbling exception
+		// needs to cause a job failure.
+
+		final Execution exec = executionGraph.getRegisteredExecutions().get(taskExecutionId);
+		if (exec == null || exec.getState() != ExecutionState.RUNNING) {
+			throw new FlinkException("Task is not running in the scheduler");
```

Comment:
The task is (or was) running on the TaskManager but was unknown on the JobManager. I think this can happen only if a failure / recovery was concurrently in progress and the event was still being in flight (delivered).

I would say that this indicates an inconsistent situation and should lead to task failure on the TaskManager.